Tasks/CopyFiles/copyfiles.ts

```diff
@@ -227,16 +226,63 @@ if (files.length > 0) {
             var targetDir = path.dirname(targetPath);
                            
             if (!createdFolders[targetDir]) {
-                tl.debug("Creating folder " + targetDir);
                 tl.mkdirP(targetDir);
                 createdFolders[targetDir] = true;
             }
 
-            if (tl.exist(targetPath) && tl.stats(targetPath).isFile() && !overWrite) {
-                console.log(tl.loc('FileAlreadyExistAt', file, targetPath));
+            // stat the target
+            let targetStats: tl.FsStats;
+            if (!cleanTargetFolder) { // skip if clean
+                try {
+                    targetStats = tl.stats(targetPath);
+                }
+                catch (err) {
+                    if (err.code != 'ENOENT') {
+                        throw err;
+                    }
+                }
+            }
+
+            // validate the target is not a directory
+            if (targetStats && targetStats.isDirectory()) {
+                throw new Error(tl.loc('TargetIsDir', file, targetPath));
+            }
+
+            if (!overWrite) {
```

Comment:
This `(!overwrite)` block is similar to the previous logic, although I removed the `-f` parameter. When the target file doesn't exist, it shouldn't be required.